gateway_code/open_nodes/common/node_segger.py

```diff
@@ -0,0 +1,139 @@
+        self.segger = self.SEGGER_CLASS.from_node(self)
+
+    @property
+    def programmer(self):
+        """Returns the segger instance of the open node."""
+        return self.segger
+
+    def clear_serial(self):
+        """Clear serial link by flushing the input buffer."""
+        try:
+            ser = serial.Serial(self.TTY, self.BAUDRATE)
+        except serial.serialutil.SerialException:
+            LOGGER.error("No serial port found")
+            return 1
+        ser.reset_input_buffer()
+        ser.close()
+        return 0
+
+    @logger_call("Node Segger: Setup of segger node")
+    def setup(self, firmware_path):
+        """ Flash open node, create serial redirection """
+        ret_val = 0
+
+        common.wait_no_tty(self.TTY)
+        ret_val += common.wait_tty(self.TTY, LOGGER)
+        ret_val += self.flash(firmware_path)
+        ret_val += self.serial_redirection.start()
+        return ret_val
+
+    @logger_call("Node Segger: teardown of segger node")
+    def teardown(self):
+        """ Stop serial redirection and flash idle firmware """
+        ret_val = 0
+        # ON may have been stopped at the end of the experiment.
+        # And then restarted again in cn teardown.
+        # This leads to problem where the TTY disappears and reappears during
+        # the first 2 seconds. So let some time if it wants to disappear first.
+        common.wait_no_tty(self.TTY)
+        ret_val += common.wait_tty(self.TTY, LOGGER)
+        # cleanup debugger before flashing
+        ret_val += self.debug_stop()
+        ret_val += self.serial_redirection.stop()
+        ret_val += self.flash(None)
+        return ret_val
+
+    @logger_call("Node Segger: flash of segger node")
+    def flash(self, firmware_path=None, binary=False, offset=0):
+        """ Flash the given firmware on segger node
+
+        :param firmware_path: Path to the firmware to be flashed on `node`.
+                              If None, flash 'idle' firmware.
+        :param binary: if True, flashes a binary file
+        :param offset: the offset at which to flash the binary file
+        """
+        firmware_path = firmware_path or self.FW_IDLE
+        LOGGER.info('Flash firmware on Segger: %s', firmware_path)
+        ret_val = self.segger.flash(firmware_path, binary, offset)
+        return ret_val
+
+    @logger_call("Node Segger: reset of segger node")
+    def reset(self):
+        """ Reset the segger node using jtag """
+        LOGGER.info('Reset segger node')
+        return self.segger.reset()
+
+    def debug_start(self):
+        """ Start segger node debugger """
+        LOGGER.info('segger node debugger start')
+        return self.segger.debug_start()
+
+    def debug_stop(self):
+        """ Stop segger node debugger """
+        LOGGER.info('segger node debugger stop')
+        return self.segger.debug_stop()
+
+    @staticmethod
+    def status():  # pylint: disable=arguments-differ
```

Comment:
Instead of silencing, just use the right interface:

```suggestion
    def status(self):
```